NetKet/Graph/abstract_graph.hpp

```diff
@@ -54,7 +54,7 @@ class AbstractGraph {
   /**
   Returns the graph edges.
   */
-  virtual std::vector<Edge> const &Edges() const noexcept = 0;
+  virtual std::vector<Edge> Edges() const noexcept = 0;
```

Comment:
Why? I made `Edges()` return a reference on purpose.

Reply:
Yes but how is this reference handled in python? What if the graph goes out of scope?

Reply:
The way I had it implemented :) Using iterables. One can always use `list` it Python to construct a list from the edges and the graph can be safely garbage-collected. Also, `edges` uses `keep_alive`, so stuff like this works:
```python
>>> import gc
>>> edges = g.edges()
>>> g = None
>>> gc.collect()
23
>>> list(edges)
[[0, 1], [1, 2], [2, 3], [0, 3]]
```